advanced-achievements-plugin/src/main/java/com/hm/achievement/placeholder/AchievementPlaceholderHook.java

```diff
@@ -47,36 +44,71 @@ public boolean persist() {
 
 	@Override
 	public String onPlaceholderRequest(Player p, String identifier) {
-		if ("total_achievements".equalsIgnoreCase(identifier)) {
-			return Integer.toString(namesToDisplayNames.size());
+		if (identifier.equalsIgnoreCase("total_achievements")) {
+			return getTotalAchievements();
 		}
 
-		if (p != null) {
-			UUID uuid = p.getUniqueId();
-			if ("achievements".equalsIgnoreCase(identifier)) {
-				return Integer.toString(cacheManager.getPlayerTotalAchievements(uuid));
-			}
+		if (p == null) {
+			return null;
+		}
 
-			if ("achievements_percentage".equalsIgnoreCase(identifier)) {
-				return String.format("%.1f%%", 100 * (double) cacheManager.getPlayerTotalAchievements(uuid)
-						/ namesToDisplayNames.size());
-			}
+		UUID uuid = p.getUniqueId();
 
-			for (NormalAchievements category : NormalAchievements.values()) {
-				if (category.toString().equalsIgnoreCase(identifier)) {
-					long statistic = cacheManager.getAndIncrementStatisticAmount(category, uuid, 0);
-					// If played time, convert from millis to hours and display one decimal.
-					return category == NormalAchievements.PLAYEDTIME ? String.format("%.1f", statistic / 3600000.0)
-							: Long.toString(statistic);
-				}
+		if (identifier.equalsIgnoreCase("achievements")) {
+			return getAchievements(uuid);
+		}
+
+		if (identifier.equalsIgnoreCase("achievements_percentage")) {
+			return getAchievementsPercentage(uuid);
+		}
+
+		if (identifier.toLowerCase().startsWith("unlocked_")) {
+			return unlockedPlaceholder(uuid, identifier.replaceFirst("(?i)unlocked_", ""));
```

Comment:
You could use `StringUtils.replaceOnceIgnoreCase(identifier, "unlocked_", "")`. Feel free to leave as is though.